src/ares/AresHandler.ts

```
@@ -310,6 +335,67 @@ function randomlyPlaceHazard(game: Game, tileType: TileType, direction: 1 | -1)
     return space;
 }
 
+
+function makeSevere(game: Game, from: TileType, to: TileType) {
+    game.board.spaces
+        .filter((s) => s.tile?.tileType === from)
+        .forEach((s) => {
+            AresHandler.putHazardAt(s, to);
+        });
+
+    game.log("${0} have upgraded to ${1}", b => b.string(tileTypeAsString(from)).string(tileTypeAsString(to)));
+}
+
+function testConstraint(constraint: IHazardConstraint, testValue: number, cb: () => void) {
+    if (!constraint.available) {
+        return;
+    }
+    if (testValue >= constraint.threshold) {
+        cb();
+        constraint.available = false;
+    }
+}
+
+function testToRemoveDustStorms(game: Game, player: Player) {
+    testConstraint(
+        game.aresData!.hazardData.removeDustStormsOceanCount,
```

As you have done the work to know `game.aresData` is defined prior to calling this I would pass in `aresData` as an argument. At that point no `!` is needed.

---

The compiler needs the assurance, unfortunately.

What would be wrong with an assert that throws an error instead of logging it? And only under those conditions, use !? Probably not, but I thought I'd ask.

---

Throwing an error makes sense. As we saw yesterday with the outage if we don't end up catching the error it takes down the node process. We _should_ be catching these errors when thrown through the server though. Can probably look into a better handler on uncaught errors.